components/movementsensor/adxl345/adxl345.go

```diff
@@ -232,6 +331,123 @@ func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error
 	return handle.WriteByteData(ctx, register, value)
 }
 
+func getByteFromInterrupts(interrupts []string) byte {
+	var register byte
+	for _, interrupt := range interrupts {
+		register += interruptBitPosition[interrupt]
+	}
+	return register
+}
+
+func (adxl *adxl345) enableInterrupts(ctx context.Context) error {
+	register := getByteFromInterrupts(adxl.interruptsEnabled)
+	return adxl.writeByte(ctx, IntEnable, register)
+}
+
+func (adxl *adxl345) setRelevantRegisters(ctx context.Context) error {
+	if len(adxl.interruptsEnabled) == 0 {
+		return nil
+	}
+	relevantRegisters := adxl.getRelevantRegisterValues()
+	for key := range defaultRegisterValues {
+		val, ok := relevantRegisters[key]
+		if ok {
+			err := adxl.writeByte(ctx, key, val)
+			if err != nil {
+				return errors.Errorf("Unable to write byte at register %x", key)
```

Comment:
Same suggestion as before: `err` describes what went wrong, so return that instead of creating a new one.